### PR TITLE
Add Vola Tab Report Binder script

### PR DESCRIPTION
**Pull request by szero**
When you have report tab bound to a given room and lain decides to
nuke your connection, the report tab will be bound to a random room
you have currently open, that regains connection first. This script
will bind the report room to one of your choosing by typing `/bindreports`
command in chat. The report room will be set even after browser restart.

### CODE REVIEW

File: volatabreportbinder.user.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// @grant        none
+// @run-at       document-start
+// ==/UserScript==
+/* globals dry */
+dry.once("dom", () => {
+  'use strict';
+  console.log("running", GM.info.script.name, GM.info.script.version, dry.version);
+  const ke = new KeyboardEvent("keydown", {
+    key: "Enter", keyCode: "13"
+  });
```
Comment (RealDolos) [inline volatabreportbinder.user.js]: I cri

File: volatabreportbinder.user.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  const bindtab = "bindmehere";
```
Comment (RealDolos) [inline volatabreportbinder.user.js]: I cri moar

File: volatabreportbinder.user.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  const ci = document.getElementById("chat_input");
```
Comment (RealDolos) [inline volatabreportbinder.user.js]: AND MOAR!

File: volatabreportbinder.user.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    bindreports() {
+      localStorage.setItem(bindtab, window.location.href);
+      dry.appendMessage("Binder", "Reports are bound to this tab now");
+      ci.value = "/reports";
+      ci.dispatchEvent(ke);
+    }
+  }();
+  new class extends dry.MessageFilter {
+    showMessage(orig, nick, message, options) {
+      let tabbind = localStorage.getItem(bindtab);
```
Comment (RealDolos) [inline volatabreportbinder.user.js]: calling localstorage.getItem a lot of times, like a pleb... tsk

File: volatabreportbinder.user.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  'use strict';
+  console.log("running", GM.info.script.name, GM.info.script.version, dry.version);
+  const ke = new KeyboardEvent("keydown", {
+    key: "Enter", keyCode: "13"
+  });
+  const bindtab = "bindmehere";
+  const ci = document.getElementById("chat_input");
+  new class extends dry.Commands {
+    bindreports() {
+      localStorage.setItem(bindtab, window.location.href);
```
Comment (RealDolos) [inline volatabreportbinder.user.js]: why not use the actual roomid?

File: volatabreportbinder.user.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+      dry.appendMessage("Binder", "Reports are bound to this tab now");
+      ci.value = "/reports";
+      ci.dispatchEvent(ke);
+    }
+  }();
+  new class extends dry.MessageFilter {
+    showMessage(orig, nick, message, options) {
+      let tabbind = localStorage.getItem(bindtab);
+      if ("Log" === nick && tabbind === window.location.href) {
+        if (message[0].hasOwnProperty("value")) {
```
Comment (RealDolos) [inline volatabreportbinder.user.js]: two ifs---nested, tsk

Comment (RealDolos) [reply volatabreportbinder.user.js]: not returning early... tsk

File: volatabreportbinder.user.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+          let match = message[0].value.match(`${dry.exts.user.info.nick} joined`);
```
Comment (RealDolos) [inline volatabreportbinder.user.js]: uh, uwutm8

File: volatabreportbinder.user.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+          if (!!match && !!dry.exts.user.info.staff) {
+            ci.value = "/reports";
```
Comment (RealDolos) [inline volatabreportbinder.user.js]: see above

File: volatabreportbinder.user.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+      ci.dispatchEvent(ke);
+    }
+  }();
+  new class extends dry.MessageFilter {
+    showMessage(orig, nick, message, options) {
+      let tabbind = localStorage.getItem(bindtab);
+      if ("Log" === nick && tabbind === window.location.href) {
+        if (message[0].hasOwnProperty("value")) {
+          let match = message[0].value.match(`${dry.exts.user.info.nick} joined`);
+          if (!!match && !!dry.exts.user.info.staff) {
```
Comment (RealDolos) [inline volatabreportbinder.user.js]: .staff? not .admin? why at all?!

File: volatabreportbinder.user.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+      localStorage.setItem(bindtab, window.location.href);
+      dry.appendMessage("Binder", "Reports are bound to this tab now");
+      ci.value = "/reports";
+      ci.dispatchEvent(ke);
+    }
+  }();
+  new class extends dry.MessageFilter {
+    showMessage(orig, nick, message, options) {
+      let tabbind = localStorage.getItem(bindtab);
+      if ("Log" === nick && tabbind === window.location.href) {
```
Comment (RealDolos) [inline volatabreportbinder.user.js]: ROOMIDDDDDDD

File: volatabreportbinder.user.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+  const ke = new KeyboardEvent("keydown", {
+    key: "Enter", keyCode: "13"
+  });
+  const bindtab = "bindmehere";
+  const ci = document.getElementById("chat_input");
+  new class extends dry.Commands {
+    bindreports() {
+      localStorage.setItem(bindtab, window.location.href);
+      dry.appendMessage("Binder", "Reports are bound to this tab now");
+      ci.value = "/reports";
```
Comment (RealDolos) [inline volatabreportbinder.user.js]: herro, let's duplicate codez!

File: volatabreportbinder.user.js [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+  console.log("running", GM.info.script.name, GM.info.script.version, dry.version);
+  const ke = new KeyboardEvent("keydown", {
+    key: "Enter", keyCode: "13"
+  });
+  const bindtab = "bindmehere";
+  const ci = document.getElementById("chat_input");
+  new class extends dry.Commands {
+    bindreports() {
+      localStorage.setItem(bindtab, window.location.href);
+      dry.appendMessage("Binder", "Reports are bound to this tab now");
```
Comment (RealDolos) [inline volatabreportbinder.user.js]: no, they are not. They are bound to this location

File: volatabreportbinder.user.js [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+          let match = message[0].value.match(`${dry.exts.user.info.nick} joined`);
+          if (!!match && !!dry.exts.user.info.staff) {
+            ci.value = "/reports";
+            ci.dispatchEvent(ke);
+          }
+        }
+      }
+    }
+  }();
+});
```
Comment (RealDolos) [inline volatabreportbinder.user.js]: >missing newline

File: volatabreportbinder.user.js [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+    showMessage(orig, nick, message, options) {
+      let tabbind = localStorage.getItem(bindtab);
+      if ("Log" === nick && tabbind === window.location.href) {
+        if (message[0].hasOwnProperty("value")) {
+          let match = message[0].value.match(`${dry.exts.user.info.nick} joined`);
+          if (!!match && !!dry.exts.user.info.staff) {
+            ci.value = "/reports";
+            ci.dispatchEvent(ke);
+          }
+        }
```
Comment (RealDolos) [inline volatabreportbinder.user.js]: herro, imma bracket pyramid!

**Review by RealDolos — CHANGES_REQUESTED** [review]
file name: admin.reportsbinder.user.js or something plox

**Comment by szero** [comment]
Review changes didn't auto-address themselves on push because I changed the filename, lul.

**Pull request merged**